### ISSUE TITLE
Possible to use —download-missing without exporting?

### ISSUE THREAD

**Issue by stibbzy** [RhetTbull/osxphotos#1662]
I have a very simple use case that I’m wondering if this tool will support. 
My MacBook battery is dead and I must always have it plugged in. Unfortunately, the Photos app sees this as a low power situation and won’t sync with iCloud automatically. I can force a sync for one hour by clicking a prompt in the Photos app, but I want to automate this. 

Is it possible to use this tool’s —download-missing feature to force an iCloud sync, but not actually export any photos?

Thanks!

**Comment by RhetTbull** [comment]
Yes, there's a script in the [examples]() directory that does just this. You can run it directly without first downloading it via:

`osxphotos run https://raw.githubusercontent.com/RhetTbull/osxphotos/main/examples/force_download.py` 

It's not guaranteed that Photos won't then immediately delete the download image which it might do if you have "optimize Mac storage" enabled.

**Comment by stibbzy** [comment]
Ahh thank you so much! I’ll give it a go!
I don’t have optimised storage on.

**Comment by stibbzy** [comment]
@RhetTbull I had a chance to try this last night. While the script ran, i watched the Photos app start syncing with iCloud, but then immediately stop and return to saying syncing was paused due to low power.
The script kept going, and said it downloaded 147 photos, but there was also lots of messages of skipped photos. No additional photos were displayed in the Photos app. I've made sure that Terminal has permissions for Photos in the Security and Privacy settings. Here's the output:
```shell
Downloading photo D5C1FB3A-0876-4851-BD25-9578BDBB8D6D.JPG
Skipping missing original photo D5C1FB3A-0876-4851-BD25-9578BDBB8D6D.JPG (F4BA9BAB-EB27-4B77-B13B-244EA7247CFD)
Downloading photo C1EE77B5-3A91-4C80-8C67-D74EEEABB3FA.JPG
Skipping missing original photo C1EE77B5-3A91-4C80-8C67-D74EEEABB3FA.JPG (533B5A1A-CC55-4DFC-A1B1-4A31EF784CC1)
Downloading photo IMG_3520.jpeg
Skipping missing original photo IMG_3520.jpeg (BEC441F6-FDE3-4D98-8A9E-9B44F1F2F9ED)
Downloading photo 191C9AEA-6516-4219-94FA-A7A7B7FDFAAF.JPG
Skipping missing original photo 191C9AEA-6516-4219-94FA-A7A7B7FDFAAF.JPG (3FEA63AE-DE78-4406-89C6-578B5F24DBEA)
Downloaded 147 photos
```

**Comment by RhetTbull** [comment]
Sorry, not sure what the issue is. If Photos fails to download the photo then osxphotos will skip it and report it missing. Perhaps try running it again? How many missing photos are you expecting (that is, photos left to download from the cloud)?

**Comment by stibbzy** [comment]
There should be a couple of hundred. 
I did try it again, but unfortunately I had the same result. 
I might just have to purchase a used Mac mini or something to overcome this dead battery issue!

**Comment by uwe-schwarz** [comment]
I was just searching for a functionality like this and found this post (thanks for the example script!).

> Downloading photo D5C1FB3A-0876-4851-BD25-9578BDBB8D6D.JPG
> Skipping missing original photo D5C1FB3A-0876-4851-BD25-9578BDBB8D6D.JPG (F4BA9BAB-EB27-4B77-B13B-244EA7247CFD)

I looked into this a while back, because not all of my photos got exported correctly and there was always a gap. For me those photos are from the shared library, but not mine. I haven't looked into this further, because I'm backuping the library from both sides (my partner and myself) and the photos will only be shared between us too.

**Comment by RhetTbull** [comment]
If you run the command `osxphotos debug-dump --dump photos --uuid D5C1FB3A-0876-4851-BD25-9578BDBB8D6D > dump.txt` and send me the output I'm happy to take a look. (Replace `D5C1FB3A-0876-4851-BD25-9578BDBB8D6D` with a UUID from the message you're seeing for your photos). The most likely cause is the photos are syndicated (shared via Messages) but not yet saved to the library. Version 0.72.0 of osxphotos (latest version) will ignore these when doing the export. If they are shared photos that have been saved to the shared library and they're not downloading there may be another problem.

**Comment by uwe-schwarz** [comment]
dump is here: https://uwe.pastes.sh/1750075737387723182 (I removed all location and classification data)

```
    'syndicated': True,
    'saved_to_library': False,
```

**Comment by RhetTbull** [comment]
```
 'syndicated': True,
    'saved_to_library': False,
```

Syndicated photos not saved to the library aren't really in the library (even though they're shown in Photos) and so osxphotos can't export these. Please update to the latest version of osxphotos and see if you still have this problem.

**Comment by uwe-schwarz** [comment]
> Syndicated photos not saved to the library aren't really in the library (even though they're shown in Photos) and so osxphotos can't export these. Please update to the latest version of osxphotos and see if you still have this problem.

I'm not having a problem here, I was just commenting on an old issue with the example script. I haven't looked into it yet, but I guess it's an easy fix to include "non syndicated" in the example script: https://github.com/RhetTbull/osxphotos/blob/main/examples/force_download.py.

**Comment by qkrrmsdud1** [comment]
> Syndicated photos not saved to the library aren't really in the library (even though they're shown in Photos) and so osxphotos can't export these.

Is there a section somewhere in the docs or readme listing the types of photos that osxphotos can't fully export yet?

**Comment by RhetTbull** [comment]
@PANICBUTT0N no there isn't. I believe unsaved syndicated photos are the only types of photos that cannot be exported by OSXPhotos. There's a problem with shared albums on macOS 26 but the shared photos themselves can be exported (must use --download-missing)
